src/app/market/sell/add-item/add-item.component.ts

```diff
@@ -315,6 +315,14 @@ export class AddItemComponent implements OnInit, OnDestroy {
     return this.itemFormGroup.valid || this.snackbar.open('Invalid Listing');
   }
 
+  inputValidate(event: any) {
+    // Special character validation
+    const keys = ['-', 'e', 'E', '+']
```

Comment:
Could we maybe define this array outside of the function? As a property on the class, perhaps? Its trivial reasoning, but technically we re-create this array everytime the function is invoked, and so we are recreating the array on every key press. It really is trivial in this specific case, but in general for performance reasons, the array does not change between invocations and so it can be created once at the creation of the component.